internal/app/machined/pkg/controllers/config/acquire_test.go

```diff
@@ -0,0 +1,455 @@
+	suite.Assert().Equal(
+		[]platform.Event{
+			{
+				Type:    platform.EventTypeConfigLoaded,
+				Message: "Talos machine config loaded successfully.",
+			},
+		},
+		suite.platformEvent.getEvents(),
+	)
+}
+
+func (suite *AcquireSuite) TestFromDiskFailure() {
+	suite.Require().NoError(os.WriteFile(suite.configPath, append([]byte("aaa"), suite.completeMachineConfig...), 0o644))
+
+	suite.triggerAcquire()
+
+	suite.AssertWithin(time.Second, 10*time.Millisecond, func() error {
+		if len(suite.platformEvent.getEvents()) == 0 || len(suite.eventPublisher.getEvents()) == 0 {
+			return retry.ExpectedErrorf("no events received")
+		}
+
+		return nil
+	})
+
+	ev := suite.platformEvent.getEvents()[0]
+	suite.Assert().Equal(platform.EventTypeFailure, ev.Type)
+	suite.Assert().Equal("Error loading and validating Talos machine config.", ev.Message)
+	suite.Assert().Equal("failed to load config from STATE: unknown keys found during decoding:\naaaversion: v1alpha1 # Indicates the schema used to decode the contents.\n", ev.Error.Error())
+
+	suite.Assert().Equal(&machineapi.ConfigLoadErrorEvent{
+		Error: "failed to load config from STATE: unknown keys found during decoding:\naaaversion: v1alpha1 # Indicates the schema used to decode the contents.\n",
+	}, suite.eventPublisher.getEvents()[0])
+}
+
+func (suite *AcquireSuite) TestFromDiskToMaintenance() {
+	suite.Require().NoError(os.WriteFile(suite.configPath, suite.partialMachineConfig, 0o644))
+
+	suite.triggerAcquire()
+
+	cfg, ok := channel.RecvWithContext(suite.Ctx(), suite.configSetter.cfgCh)
+	suite.Require().True(ok)
+
+	suite.Require().Equal(cfg.SideroLink().APIUrl().Host, "siderolink.api")
+
+	suite.injectViaMaintenance(suite.completeMachineConfig)
+
+	cfg = suite.waitForConfig()
+	suite.Require().Equal(cfg.Cluster().Name(), suite.clusterName)
+
+	suite.Assert().Equal(
+		[]proto.Message{
+			&machineapi.TaskEvent{
+				Action: machineapi.TaskEvent_START,
+				Task:   "runningMaintenance",
+			},
+			&machineapi.TaskEvent{
+				Action: machineapi.TaskEvent_STOP,
+				Task:   "runningMaintenance",
+			},
+		},
+		suite.eventPublisher.getEvents(),
+	)
+	suite.Assert().Equal(
+		[]platform.Event{
+			{
+				Type:    platform.EventTypeActivate,
+				Message: "Talos booted into maintenance mode. Ready for user interaction.",
+			},
+			{
+				Type:    platform.EventTypeConfigLoaded,
+				Message: "Talos machine config loaded successfully.",
+			},
+		},
+		suite.platformEvent.getEvents(),
+	)
+}
+
+func (suite *AcquireSuite) TestFromPlatform() {
+	suite.platformConfig.configuration = suite.completeMachineConfig
+	suite.platformConfig.err = nil
+
+	suite.triggerAcquire()
+
+	cfg := suite.waitForConfig()
+	suite.Require().Equal(cfg.Cluster().Name(), suite.clusterName)
+
+	suite.Assert().Empty(suite.eventPublisher.getEvents())
+	suite.Assert().Equal(
+		[]platform.Event{
+			{
+				Type:    platform.EventTypeConfigLoaded,
+				Message: "Talos machine config loaded successfully.",
```

Comment:
maybe we update the even with the source too, then message becomes more clear from where the config was loaded from

Reply:
I tried to keep things as compatible as it can be, idk if Equinix is using this event literally? it will be only visible on the EM console at the moment iirc

Reply:
i meant the general message that goes to dmesg, something like: `Talos machine config loaded successfully from disk` or `Talos machine config loaded successfully from platform`

Reply:
ah, this makes sense... it might be actually multiple sources, need to think about it...

Reply:
added sources